dark/diamond/conversion.py

```diff
@@ -0,0 +1,286 @@
+                            'frame': int(qframe),
+                            'query': qseq,
+                            'query_start': int(qstart),
+                            'query_end': int(qend),
+                            'sbjct': sseq,
+                            'sbjct_start': int(sstart),
+                            'sbjct_end': int(send),
+                        }
+                        alignment = {
+                            'hsps': [hsp],
+                            'length': int(slen),
+                            'title': sseqid,
+                        }
+                        record['alignments'].append(alignment)
+                    else:
+                        # We have already seen this subject, so this is another
+                        # HSP in an already existing alignment.
+                        hsp = {
+                            'bits': float(bitscore),
+                            'expect': float(evalue),
+                            'frame': int(qframe),
+                            'query': qseq,
+                            'query_start': int(qstart),
+                            'query_end': int(qend),
+                            'sbjct': sseq,
+                            'sbjct_start': int(sstart),
+                            'sbjct_end': int(send),
+                        }
+                        for alignment in record['alignments']:
+                            if alignment['title'] == sseqid:
+                                alignment['hsps'].append(hsp)
+                else:
+                    # all alignments for the previous qseqid have been seen. If
+                    # this is not the first record, yield it and start
+                    # accumulating the alignments for the current qseqid.
+                    if previousQseqid is not None:
+                        subjectsSeen = {}
+                        yield record
+                        record = {}
+
+                    # start building up the new record
+                    subjectsSeen = {sseqid}
+                    hsp = {
+                        'bits': float(bitscore),
+                        'expect': float(evalue),
+                        'frame': int(qframe),
+                        'query': qseq,
+                        'query_start': int(qstart),
+                        'query_end': int(qend),
+                        'sbjct': sseq,
+                        'sbjct_start': int(sstart),
+                        'sbjct_end': int(send),
+                    }
+                    alignment = {
+                        'hsps': [hsp],
+                        'length': int(slen),
+                        'title': sseqid,
+                    }
+                    record['alignments'] = [alignment]
+                    record['query'] = qseqid
+
+                    previousQseqid = qseqid
+            # Yield the last record.
+            yield record
```

Comment:
Do you need to say `if record`?  I think you'd at least need that in the case of totally empty input.